src/modules/appointment/appointment.service.ts

```diff
@@ -0,0 +1,143 @@
+    }
+  }
+
+  // TODO: can be used for getting appointments for today, week and month
+  async getAppointmentsByDoctorId(id: number): Promise<Appointment[]> {
+    try {
+      const doctor = await this.doctorService.getDoctorByID(id);
+
+      const queryBuilder: SelectQueryBuilder<Appointment> =
+        this.appointmentRepository.createQueryBuilder('appointment');
+      queryBuilder.where(
+        'appointment.localDoctorId = :doctorId OR appointment.remoteDoctorId = :doctorId',
+        { doctorId: doctor.id },
+      );
+      queryBuilder.leftJoinAndSelect('appointment.localDoctor', 'localDoctor');
+      queryBuilder.leftJoinAndSelect(
+        'appointment.remoteDoctor',
+        'remoteDoctor',
+      );
+      queryBuilder.leftJoinAndSelect('appointment.patient', 'patient');
+      queryBuilder.select([
+        'appointment.id',
+        'appointment.startTime',
+        'appointment.endTime',
+        'appointment.zoomLink',
+        'localDoctor.id',
+        'remoteDoctor.id',
+        'patient.id',
+      ]);
+
+      return await queryBuilder.getMany();
+    } catch (err) {
+      throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  // TODO: can be used for getting appointments for today, week and month
+  async getAppointmentsByPatientId(id: number): Promise<Appointment[]> {
+    try {
+      const patient = await this.patientService.getPatientById(id);
+
+      const queryBuilder: SelectQueryBuilder<Appointment> =
+        this.appointmentRepository.createQueryBuilder('appointment');
+      queryBuilder.where('appointment.patientId = :patientId', {
+        patientId: patient.id,
+      });
+      queryBuilder.leftJoinAndSelect('appointment.localDoctor', 'localDoctor');
+      queryBuilder.leftJoinAndSelect(
+        'appointment.remoteDoctor',
+        'remoteDoctor',
+      );
+      queryBuilder.leftJoinAndSelect('appointment.patient', 'patient');
+      queryBuilder.select([
+        'appointment.id',
+        'appointment.startTime',
+        'appointment.endTime',
+        'appointment.zoomLink',
+        'localDoctor.id',
+        'remoteDoctor.id',
+        'patient.id',
+      ]);
+
+      return await queryBuilder.getMany();
+    } catch (err) {
+      throw new HttpException(`${err}`, HttpStatus.INTERNAL_SERVER_ERROR);
+    }
+  }
+
+  async getPatientsByDoctorIdAppointments(id: number): Promise<Patient[]> {
+    try {
+      const doctor = await this.doctorService.getDoctorByID(id);
+
+      const appointments = await this.appointmentRepository
+        .createQueryBuilder('appointment')
+        .leftJoinAndSelect('appointment.patient', 'patient')
+        .where(
+          'appointment.localDoctorId = :id OR appointment.remoteDoctorId = :id',
+          { id: doctor.id },
+        )
+        .getMany();
+
+      const patients: Patient[] = appointments.map(
+        (appointment) => appointment.patient,
+      );
```

Comment:
can we find other way, map through each appointment is expencive

Reply:
rewrited with subQuery